plugin/trino-hive/src/main/java/io/trino/plugin/hive/aws/athena/PartitionProjectionService.java

```diff
@@ -186,23 +187,19 @@ public Map<String, String> getPartitionProjectionHiveTableProperties(ConnectorTa
                             String.class::cast);
                 });
 
-        Optional<Boolean> projectionEnabledProperty = Optional.ofNullable((Boolean) tableMetadata.getProperties().get(PARTITION_PROJECTION_ENABLED));
-        if (projectionEnabledProperty.isPresent()) {
-            // If enable property in place we validate settings.
-            List<String> partitionColumnNames = getPartitionedBy(tableMetadata.getProperties());
-            createPartitionProjection(
-                    projectionEnabledProperty.orElse(false),
-                    tableMetadata.getColumns().stream()
-                            .filter(columnMetadata -> partitionColumnNames.contains(columnMetadata.getName()))
-                            .collect(toImmutableMap(
-                                    columnMetadata -> columnMetadata.getName(),
-                                    columnMetadata -> columnMetadata.getType())),
-                    tableMetadata.getColumns()
-                            .stream()
-                            .map(ColumnMetadata::getName)
-                            .collect(toImmutableList()),
-                    metastoreTableProperties);
-        }
+        // We initialize partition projection to validate properties.
+        List<String> partitionColumnNames = getPartitionedBy(tableMetadata.getProperties());
```

Comment:
Do we need to change this part of code back and forth many times - hard to review.

Reply:
I've added it as separate commit so it's easier to discuss adding this feature or not. Now i think i can squash last 3 commits. As it's whole Partition Projection